iothub/device/src/ConnectionStateChangesReason.cs

```diff
@@ -4,71 +4,71 @@
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// Connection status change reason supported by the client.
+    /// Connection state change reason supported by the client.
     /// </summary>
-    public enum ConnectionStatusChangeReason
+    public enum ConnectionStateChangesReason
     {
         /// <summary>
         /// The client is connected, and ready to be used.
-        /// <para>This is returned with a connection status of <see cref="ConnectionStatus.Connected"/>.</para>
+        /// <para>This is returned with a connection state of <see cref="ConnectionState.Connected"/>.</para>
         /// </summary>
-        Connection_Ok,
+        ConnectionOk,
 
         /// <summary>
         /// The SAS token associated with the client has expired, and cannot be renewed.
         /// The supplied credentials need to be fixed before a connection can be established.
         /// <para>NOTE: This is currently not used in the client library.</para>
         /// </summary>
-        Expired_SAS_Token,
+        ExpiredSasToken,
```

Comment:
I think this would be a good opportunity for us to discuss if we want to use this.

Reply:
We could expand the reason values associated with "Disconnected" state.

Reply:
Oh, because currently not implemented?

Reply:
Let's move that to a different task/PR. Good idea to pursue, though. Perhaps you'll pick it up, Abhipsa?